### ISSUE TITLE
[feat req] Upgrade action to use node20

### ISSUE THREAD

**Issue by erikburt** [actions/download-artifact#230]
### What would you like to be added?

Upgrade action to use node20

### Why is this needed?

https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

> Node 16 has reached its [end of life](https://github.com/nodejs/Release/#end-of-life-releases), prompting us to initiate its deprecation process for GitHub Actions. Our plan is to transition all actions to run on Node 20 by Spring 2024. We will actively monitor the migration's progress and gather community feedback before finalizing the transition date. Starting October 23rd, workflows containing actions running on Node 16 will display a warning to alert users about the upcoming migration.
> ## What you need to do
> ### For Actions maintainers
> Modify your actions to run on Node 20 instead of Node 16. For guidance, refer to the [Actions configuration settings ](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-javascript-actions).

**Comment by vdovhanych** [comment]
This is more of a bug than a feature request, as the latest selfhosted runner refuses to run with the node16 environment. It would be nice to update it.

**Comment by newAM** [comment]
It is easy to point the action at a different repository that uses node20 as a temporary hack.  I'm using the code from #238 for my self-hosted runners, works great.

**Comment by vdovhanych** [comment]
Yes, I know, but it should also be updated here. It will fail for other users of this action, especially since we already have working versions, and you do not need to make any other changes.

**Comment by per1234** [comment]
This was resolved by https://github.com/actions/download-artifact/pull/233, which is released in version 4.0.0 of the action.

**Comment by erikburt** [comment]
^ Closing
